### PR TITLE
rp-info currency format

### LINKED ISSUE #340: rp-info value returned requires consistent zcn currency format
(But equally applies to any other command returning zcn currency value)

example output from `zbox rp-info`
`  		balance :             3.000 ZCN`
example output from `zbox rp-info --json`
`{"balance":30000000000}`

compare with:-

example output from `zwallet getbalance`
`Balance: 7.000 ZCN (1.22 USD)`
example output from `zwallet getbalance -json`
`{"fmt":"7.000 ZCN","usd":"1.223488","zcn":"7.000000"}`

I think zwallet getbalance is much better format and pretty much agreed upon, however if we require microtoken balance (sas) we should add that to the JSON output of the --json command, e.g.
`{"fmt":"7.000 ZCN","usd":"1.223488","zcn":"7.000000","sas":"70000000000"}`

### PR DESCRIPTION
**Pull request by SinanEkm**
A brief description of the changes in this PR: 
Some changes have been made to the rp-info method. After the changes, rp-info method and --json option returning an output more like the ./zwallet getbalance method's output. 

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zboxcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- 0chain: 
- blobber:
- gosdk:
- system_test:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by Kishan-Dhakan** [comment]
@SinanEkm what's the status of this PR?

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | 340-rp-info-value-returned-requires-consistent-zcn-currency-format  |
| gosdk  | staging  |
| zboxcli  | 340-rp-info-value-returned-requires-consistent-zcn-currency-format  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | model.go-ReadPoolInfo-change  |
| gosdk  | staging  |
| zboxcli  | 340-rp-info-value-returned-requires-consistent-zcn-currency-format  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | model.go-ReadPoolInfo-change  |
| gosdk  | staging  |
| zboxcli  | 340-rp-info-value-returned-requires-consistent-zcn-currency-format  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — COMMENTED** [review]
lgtm, is it the only pool where we have such a problem? does it work for stakepool and writepool correctly?

**Comment by SinanEkm** [comment]
> lgtm, is it the only pool where we have such a problem? does it work for stakepool and writepool correctly?

do you mean formatting @dabasov ? If so, I checked sp-info command and it doesn't show USD value of the balance, too. I am not sure about writepool, is there wp-info command or any such thing that is equivalent to it?

File: cmd/readpool.go [hunk 1]
```diff
@@ -45,16 +39,28 @@ var rpInfo = &cobra.Command{
...
+		}
+
 		if info.Balance == 0 {
 			fmt.Println("no tokens locked")
 			return
 		}
 
 		if doJSON {
-			util.PrintJSON(info)
+			kk := map[string]interface{}{
```
Comment (Kishan-Dhakan) [inline cmd/readpool.go]: Please name the variable to something meaningful other than kk

File: cmd/readpool.go [hunk 2]
```diff
@@ -45,16 +39,28 @@ var rpInfo = &cobra.Command{
...
+				"usd": usd,
+				"zcn": bt,
+				"fmt": info.Balance,
+			}
+
+			util.PrintJSON(kk)
 			return
 		}
-		printReadPoolInfo(info)
+		fmt.Printf("\nBalance: %v (%.2f USD)\n", info.Balance, usd)
```
Comment (Kishan-Dhakan) [inline cmd/readpool.go]: Sugestion: can make the text: `Read pool Balance: %v (%.2f USD)\n", info.Balance, usd`

**Review by Kishan-Dhakan — CHANGES_REQUESTED** [review]

File: cmd/readpool.go (hunk 1 above)
Comment (SinanEkm) [reply cmd/readpool.go]: Oh sorry, I forgot that during the development. Changing it.

File: cmd/readpool.go (hunk 2 above)
Comment (SinanEkm) [reply cmd/readpool.go]: Yeah sure.

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | model.go-ReadPoolInfo-change  |
| gosdk  | staging  |
| zboxcli  | 340-rp-info-value-returned-requires-consistent-zcn-currency-format  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by Kishan-Dhakan — APPROVED** [review]

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by dabasov** [comment]
> > lgtm, is it the only pool where we have such a problem? does it work for stakepool and writepool correctly?
> 
> do you mean formatting @dabasov ? If so, I checked sp-info command and it doesn't show USD value of the balance, too. I am not sure about writepool, is there wp-info command or any such thing that is equivalent to it?

I mean this https://github.com/0chain/zboxcli/blob/23bc1d65cfbe6f33b51537c5c50a092b45a7b544/cmd/stakepool.go
should we have consistent output here?

**Pull request merged**